cfme/common/__init__.py

```diff
@@ -429,8 +429,9 @@ def add_tag(self, item_objects, tag=None, cancel=False, reset=False):
         """
         view = navigate_to(self, 'All')
         for item in item_objects:
-            name = item if isinstance(item, six.string_types) else item.name
-            view.entities.get_entity(surf_pages=True, name=name).check()
+            kwargs = {get_entity_by: item if isinstance(
```

Comment:
@rlbabyuk I had a problem to set a tag for an image registry object because the function ```add_tag``` selects the object to tag by name. since there isn't a name column in the all view for image registry the function failed to find the required object in the function and raise an exception.
By adding the ```get_entity_by parameter``` I can select on which colume to look for the reqired item.

Reply:
Question: Does it work for grid and tile view as well, maybe better to switch to other view

Reply:
@quarckster What do you think? As I remember we get entities by name or id, as other types cannot be available in grid or tile views, or I'm mistaken here?

Reply:
It seems legit for me, I would only change `get_entity_by` variable to `keyword_name`

Reply:
@rlbabyuk Done